server/utils/scraping.js

```diff
@@ -0,0 +1,122 @@
+const puppeteer = require('puppeteer');
+
+function domainName(url) {
+	const urlBeginning = /^https?:\/\/?w+\./;
+	const ebayRe = /^\bebay\b/;
+	const amazonRe = /^\bamazon\b/;
+	const craigslistRe = /\b\.craigslist\.\b/;
+	let domain = '';
+	let domainNoEnd = url.replace(urlBeginning, '');
+
+	if (ebayRe.test(domainNoEnd)) {
+		domain = 'ebay';
+	} else if (amazonRe.test(domainNoEnd)) {
+		domain = 'amazon';
+	} else if (craigslistRe.test(domainNoEnd)) {
+		domain = 'craigslist';
+	} else {
+		domain = 'Please enter an ebay or amazon domain';
+	}
+	return domain;
+}
+const scraping = async (url) => {
+	let pageData = {};
+
+	try {
+		const browser = await puppeteer.launch({ headless: true });
+		const page = await browser.newPage();
+		await page.goto(url);
+
+		//This makes sure that the page has loaded based on one selector
+		if (domainName(url) === 'amazon') {
+			await page.waitForSelector('span#productTitle');
+
+			pageData = await page.evaluate(() => {
+				const priceBeginning = /[a-zA-Z]*/;
+				const dollarSign = /[$]/;
+				//nothing inside or outside of page.evaluate is available outside of it's closure, because it is running purely in the dom
+				let titleData = document.getElementById(`productTitle`);
+				let priceData = document.getElementById(`price_inside_buybox`);
+				let imageData = document.getElementById(`landingImage`);
+				// let descriptionData = document.getElementById(`productDescription`);
+				// let descriptionData = [...document.querySelectorAll(`#feature-bullets ul li span`)].map(
```

Comment:
One alternative to product description is - we can display `<a href="url_to_the_product" target="_blank">View Details</a>

Reply:
yeah I'd be down to do that!